internal/folders/manager.go

```diff
@@ -0,0 +1,179 @@
+)
+
+type Manager struct {
+	DbClient *db.Manager
+}
+
+func New(dbManager *db.Manager) *Manager {
+	manager := &Manager{
+		DbClient: dbManager,
+	}
+	return manager
+}
+
+// Get lists folders for current user 
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Get Folders for current User
+//	@Description	get folders for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{array}	folders.Folders
+//	@Router			/folders [get]
+func (m *Manager) Get(w http.ResponseWriter, r *http.Request) {
+	userId, err := strconv.Atoi(chi.URLParam(r, "user_id"))
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+
+	dbFolders := m.DbClient.GetFolders(userId)
+	response := Folders{
+		Folders: FromDBTypeArray(dbFolders),
+	}
+	writeJson(w, response)
+}
+
+// Create folder for current user 
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Create Folder for current User
+//	@Description	new folder for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{object}	folders.Folder
+//	@Router			/folders [post]
+func (m *Manager) Post(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	body, _ := ioutil.ReadAll(r.Body)
+
+	folder := &Folder{}
+	err := json.Unmarshal(body, folder)
+	if err != nil {
+		writeStatus(w, http.StatusInternalServerError)
+	}
+
+	dBFolder := &db.Folder{
+		Name:   folder.Name,
+		Order:  folder.Order,
+		UserId: folder.UserId,
+	}
+
+	err = m.DbClient.CreateFolder(dBFolder)
+	if err != nil {
+		log.Print(err)
+		writeStatus(w, http.StatusInternalServerError)
+	}
+
+	writeStatus(w, http.StatusCreated)
+}
+
+// Get folder by ID
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Get folder by ID
+//	@Description	get current folder for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{object}	folders.Folder
+//	@Router			/folders/{id} [get]
+func (m *Manager) GetByID(w http.ResponseWriter, r *http.Request) {
+	folderId, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		log.Printf("%v", err)
+	}
+	dbFolder := m.DbClient.GetFolderById(folderId)
+	writeJson(w, FromDBType(dbFolder))
+}
+
+// Update folder by ID
+// not done
+
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Update folder by ID
+//	@Description	update a folder for user
+//	@Tags			folders
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{object}	folders.Folder
+//	@Router			/folders/{id} [put]
+func (m *Manager) Put(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	body, _ := ioutil.ReadAll(r.Body)
+
+	folder := &Folder{}
+	err := json.Unmarshal(body, folder)
+	if err != nil {
+		writeStatus(w, http.StatusInternalServerError)
+	}
+
+	dBFolder := &db.Folder{
+		Id: folder.Id,
+		Name:   folder.Name,
+		Order:  folder.Order,
+	}
+
+	err = m.DbClient.UpdateFolder(dBFolder)
+	if err != nil {
+		log.Print(err)
+		writeStatus(w, http.StatusInternalServerError)
```

Comment:
We probably need some more specific error handling here, since one of the possible failure modes is trying to PUT to an ID that doesn't exist, and this line will cause it to return a 500 error, even though the issue is on the caller's side for attempting to modify a non-existent object.

Ideally, we'd want to distinguish between server-side errors (5xx errors) and client-side errors (4xx errors), but I'm not sure what the best way of doing that in Go is.

It looks like newerish Go has an [`errors.Is` function](https://stackoverflow.com/a/43606199) for checking an error of a known type, but I'm not exactly sure how to use it, so I may punt on this.